apps/aecore/test/aecore_suite_utils.erl

```diff
@@ -0,0 +1,517 @@
+    Flags = ["-pa ", MyDir, " -config ./" ++ ConfigFilename],
+    cmd(["(cd ", node_shortcut(N, Config),
+         " && ERL_FLAGS=\"", Flags, "\"",
+         " EPOCH_CONFIG=./data/epoch.json"
+         " RUNNER_LOG_DIR=`pwd`/log"
+         " CODE_LOADING_MODE=interactive"
+         " ./bin/epoch start)"]).
+
+stop_node(N, Config) ->
+    cmd(["(cd ", node_shortcut(N, Config),
+         " && ./bin/epoch stop)"]).
+
+mine_one_block(Node) ->
+    aecore_suite_utils:subscribe(Node, block_created),
+    StartRes = aecore_suite_utils:rpc_call(Node, aec_conductor, start_mining, []),
+    ct:log("aec_conductor:start_mining() (~p) -> ~p", [Node, StartRes]),
+    receive
+        {gproc_ps_event, block_created, Info} ->
+            StopRes = aecore_suite_utils:rpc_call(Node, aec_conductor, stop_mining, []),
+            ct:log("aec_conductor:stop_mining() (~p) -> ~p", [Node, StopRes]),
+            ct:log("block created, Info=~p", [Info]),
+            ok
+    after 30000 ->
+            StopRes = aecore_suite_utils:rpc_call(Node, aec_conductor, stop_mining, []),
+            ct:log("aec_conductor:stop_mining() (~p) -> ~p", [Node, StopRes]),
+            ct:log("timeout waiting for block event~n"
+                   "~p", [process_info(self(), messages)]),
+            error(timeout_waiting_for_block)
+    end.
+
+mine_blocks(Node, NumBlocksToMine) ->
+    aecore_suite_utils:subscribe(Node, block_created),
+    StartRes = aecore_suite_utils:rpc_call(Node, aec_conductor, start_mining, []),
+    ct:log("aec_conductor:start_mining() (~p) -> ~p", [Node, StartRes]),
+    Res = mine_blocks_loop(NumBlocksToMine),
+    StopRes = aecore_suite_utils:rpc_call(Node, aec_conductor, stop_mining, []),
+    ct:log("aec_conductor:stop_mining() (~p) -> ~p", [Node, StopRes]),
+    case Res of
+        ok ->
+            ok;
+        {error, Reason} ->
+            erlang:error(Reason)
+    end.
+
+mine_blocks_loop(0) ->
+    ok;
+mine_blocks_loop(BlocksToMine) ->
+    receive
+        {gproc_ps_event, block_created, Info} ->
+            ct:log("block created, Info=~p", [Info]),
+            mine_blocks_loop(BlocksToMine -1)
+    after 30000 ->
+            ct:log("timeout waiting for block event~n"
+                  "~p", [process_info(self(), messages)]),
+            {error, timeout_waiting_for_block}
+    end.
+
+top_dir(DataDir) ->
+    %% Split the DataDir path at "_build"
+    [Top, _] = re:split(DataDir, "_build", [{return, list}]),
+    Top.
+
+node_tuple(N) when N == dev1; N == dev2; N == dev3 ->
+    {N, node_name(N)}.
+
+node_name(N) when N == dev1; N == dev2; N == dev3 ->
+    [_,H] = re:split(atom_to_list(node()), "@", [{return,list}]),
+    list_to_atom("epoch_" ++ atom_to_list(N) ++ "@" ++ H).
+
+connect(N) ->
+    connect(N, 5),
+    report_node_config(N).
+
+subscribe(N, Event) ->
+    call_proxy(N, {subscribe, Event}).
+
+unsubscribe(N, Event) ->
+    call_proxy(N, {unsubscribe, Event}).
+
+rpc_call(Node, Module, Function, Arguments) ->
```

Comment:
Should we not leave it to the test designer which timeout to use? I prefer not to have a util function that basically calls a well known library function